### PR TITLE
physics.vector: respect local printer settings in Dyadic and Vector 

### PR DESCRIPTION
**Pull request by eric-wieser**
#### References to other Issues or PRs

Builds upon #19621, ~~marking as draft until that is merged.~~

Same idea as #19424.

#### Brief description of what is fixed or changed
Previously, Both `_sympystr` and `_latex` would ignore the settings (and subclass) of the printer passed in, instead constructing their own printer. The test added in this PR previously failed.

This follow the same idea as 7c804f0.

As a result of this changing, no vector printing customizations will be enabled until `init_vprinting` is called.

#### Release Notes

* physics.vector
  * When printing, `Vector` and `Dyadic` now respect keyword-arguments to `sstr` and `latex`. As a result, these no longer use vector printing customizations unless `init_vprinting` has been called.

### CODE REVIEW

**Comment by eric-wieser** [comment]
Would it be appropriate to call `init_vprinting()` in the doctests?

**Comment by asmeurer** [comment]
The doctests should show what we want the typical user to do, so if we want a typical user of the module to use init_vprinting, then we should use it.

**Comment by eric-wieser** [comment]
@asmeurer: My fear is that because `init_vprinting` modifies global state, it will impact all the other doctests in other files.

**Comment by eric-wieser** [comment]
Ah, looks like that's already handled.

**Comment by eric-wieser** [comment]
Updated to use `init_vprinting`, lets see if I missed any

File: doc/src/modules/physics/mechanics/masses.rst [hunk 1]
```diff
@@ -352,9 +356,9 @@ The user can then determine the kinetic energy of any number of entities of the
 system: ::
 
   >>> kinetic_energy(N, Pa)
-  2*l1**2*m*omega(t)**2
```
Comment (eric-wieser) [inline doc/src/modules/physics/mechanics/masses.rst:355]: Look at the lines above - previously the printing was inconsistent, sometime with `(t)` and sometimes without. This makes it all the same.

Removing `init_vprinting` would do the reverse and use `(t)` everywhere - but at that point, there'd be no reason to use `dynamicsymbols` in this example.

**Comment by oscarbenjamin** [comment]
Looks good

**Comment by moorepants** [comment]
I'm just noticing some of these changes in the release notes. We have traditionally pinged @sympy/mechanics for changes to sympy.physics.vector and sympy.physics.mechanics code. @eric-wieser If you want to join the maintenance of these two modules, we can add you to that team too. This change is a backwards incompatible change and I would very much appreciate being requested for reviews on these two modules before others merge in the code. We have people that maintain certain modules for that reason.

**Comment by eric-wieser** [comment]
> I'm just noticing some of these changes in the release notes. We have traditionally pinged @sympy/mechanics for changes to sympy.physics.vector and sympy.physics.mechanics code.

Apologies, I'm not too familiar with how everything works with teams within sympy.

> @eric-wieser If you want to join the maintenance of these two modules, we can add you to that team too. 

Thanks for the offer.
I don't have any particular interest in joining that team, having not used these parts of sympy much - my motivation in this PR was part of a larger push to clean up uses of the printer API sympy-wide (having spent a while trying to work out the right way to handle things in `galgebra`). My involvement in other PRs was mainly just in trying to find something easy to review.

> I would very much appreciate being requested for reviews on these two modules before others merge in the code

Have you considered setting up `CODEOWNERS` to make this automatic?

I'm happy to respond to review comments post-merge here, and follow up with a fixup PR.

**Comment by moorepants** [comment]
> I don't have any particular interest in joining that team, having not used these parts of sympy much - my motivation in this PR was part of a larger push to clean up uses of the printer API sympy-wide (having spent a while trying to work out the right way to handle things in galgebra). My involvement in other PRs was mainly just in trying to find something easy to review.

Ok. That's helpful, for sure. I'm likely more sensitive to backwards incompatible changes than most, especially for these modules as I have now have hundreds of notebooks and files that depend on it. For example, if I and my students now have to add `init_vprinting()` to a large number of existing iles to have the same behavior as the last decade, that's a bit of a pain. This PR seems to fix a non-existing issue (at least for Vector/Dyadic). I am not aware of any issues with these printers needing to work differently. Cleanups are fine, but they should maintain input/output or be properly deprecated (and discussed).

> Have you considered setting up CODEOWNERS to make this automatic?

No, I am not aware of this. I certainly can no longer make any sense of the github notifications because I get hundreds of them that are of no interest to me. Anything that could help be notified more narrowly would help. I will look into it. Thanks for the tip.

**Comment by eric-wieser** [comment]
> This PR seems to fix a non-existing issue (at least for Vector/Dyadic).

While perhaps not being an important issue, this fixes a problem where `Vector` completely ignores the `full_prec` printer setting when printing its coefficients. It also makes it respect `inv_trig_style`, etc. The test uses `full_prec` simply because that's the most obvious, but the actual bug was that all settings were ignored.

> I am not aware of any issues with these printers needing to work differently.

This patch not only fixes ignored settings, but also prevents invalid latex being emitted as in the following code

```python
>>> from sympy.physics.vector import ReferenceFrame
>>> from sympy import latex
>>> from sympy.printing.printer import Printer

>>> Printer.set_global_settings(mode='equation')

>>> N = ReferenceFrame('N')
>>> latex(2 * N.x, mode='equation')
'\\begin{equation}\\begin{equation}2\\end{equation}\\mathbf{\\hat{n}_x}\\end{equation}'
```

This last line is invalid latex.

**Comment by eric-wieser** [comment]
The changes here were motivated by the doc change in #19622 - it seemed like sympy should follow its own advice.

**Comment by moorepants** [comment]
I see that an issue you discovered motivated the change, but the backward incompatibility should follow our deprecation policy if at all possible: https://github.com/sympy/sympy/wiki/Deprecating-policy.

As an extra note, the objects in vector/ and mechanics/ are, for better or worse, not proper SymPy objects. Lots of things don't work like objects based on the sympy core types. This may have been a big design flaw, but it is was it is. This isn't the only thing that will arise like this in the codebase.

**Comment by eric-wieser** [comment]
To me, printing is a gray area for deprecations - both `Derivative(u(t), t)` and `u'` are accurate representations of the same object, and since those outputs should be consumed by humans only, it's difficult to see how they can result in a "breaking change". (Having said that, I have a project that checks notebook output in CI, and would be broken by this type of thing).

I'm not arguing that this change is immune from the policy - but more that printing is a weird area that should be called out explicitly if it is intended to be included.

Something like this came up in numpy a while back - we changed `ndarray.__repr__` (for the better), but knew that doing so would break doctests. We didn't treat it like a normal deprecation though, and instead added `np.set_print_settings(legacy='1.13')` to allow users to restore old behavior to fix their doctests. I suppose something similar could be done here, but I suspect the impact of the change is far lower.

**Comment by eric-wieser** [comment]
> As an extra note, the objects in vector/ and mechanics/ are, for better or worse, not proper SymPy objects. Lots of things don't work like objects based on the sympy core types. This may have been a big design flaw, but it is was it is. This isn't the only thing that will arise like this in the codebase.

`galgebra` suffers from exactly the same problem. Part of my motivation for the work on the printers was to make this not matter any more - the printing mechanism no longer cares if your objects are proper sympy objects or not, and no longer has a hard-coded list of supported non-proper objects.

**Comment by moorepants** [comment]
Sure, but there was no discussion in the review about deprecation or not, gray, black, or white.

**Comment by moorepants** [comment]
Funny historical note, is that when the mechanics package was designed in 2009, galgebra was used as strong influence of this code. In (my) hindsight, that wasn't the best package to mimic.

**Comment by eric-wieser** [comment]
That's actually useful info to me, thanks - since this history of galgebra is fragmented across multiple repos, I'd assumed that the design influence went in the other direction, and mechanics came first. This was actually another motivation for this change, I wanted to ensure nothing else copied the same printing design mistakes!

**Comment by moorepants** [comment]
galgebra did come first :)

**Comment by oscarbenjamin** [comment]
I guess if we merge the CODEOWNERS file in #20130 it won't be necessary to explicitly ping for this sort of thing. 

Would it make more sense for the CODEOWNERS file to refer to @sympy/mechanics rather than @moorepants?

The deprecation policy doesn't explain what is supposed to count as a "backward incompatible" change. I'm not sure if this is something that can be defined easily though. SymPy has a lot of bugs due to inconsistent use of its own internal designs but essentially any fix or cleanup will not be backwards compatible in the strictest sense.

There are parts of sympy most notably some parts of physics and matrices but also holonomic and some other places that don't use the standard `Basic` model of sympy. Each of these places then becomes a continuous source of maintenance headaches because the attempt to replicate Basic without subclassing it leads to a huge amount of duplication and is never complete.

**Comment by moorepants** [comment]
> Would it make more sense for the CODEOWNERS file to refer to @sympy/mechanics rather than @moorepants?

Sure, it wasn't clear to me if that worked after reading the Github documentation. And if it did work, would it add all 9 members of that team to the PR?

**Comment by eric-wieser** [comment]
Perhaps the easiest thing to do is just

> Sure, it wasn't clear to me if that worked after reading the Github documentation. And if it did work, would it add all 9 members of that team to the PR?

Perhaps simplest to try it and see?

**Comment by eric-wieser** [comment]
My guess is that `@sympy/mechanics` would not work, as it does not appear in the review dropdown - while `@sympy/gsoc-students` would work, as it does. I don't know what the difference is between these two pseudo-users, and why only one appears.

**Comment by moorepants** [comment]
I think you also have to have write access to the repo.

**Pull request merged**